### PR TITLE
src-cli: init at 4.0.0

### PR DESCRIPTION
**Pull request by dit7ya**
###### Description of changes
src is a command line interface to Sourcegraph
https://github.com/sourcegraph/src-cli

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/src-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  vendorSha256 = "sha256-ll/Q7hZJngOGG/Jl79KvEIHQ+ARDxFD7KHY3nSokRi0=";
+
+  subPackages = [ "cmd/src" ];
+
+  CGO_ENABLED = 0;
+
+  ldflags = [ "-X github.com/sourcegraph/src-cli/internal/version.BuildTag=${version}" ];
+
+  meta = with lib; {
```
Comment (IvarWithoutBones) [inline pkgs/development/tools/misc/src-cli/default.nix:25]: Could you add `changelog = "https://github.com/sourcegraph/src-cli/blob/${version}/CHANGELOG.md";`?

File: pkgs/development/tools/misc/src-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-cd96ddFtPOMATI0MEMFH399/y7SZeASetVhvRt0P38A=";
+  };
+
+  vendorSha256 = "sha256-ll/Q7hZJngOGG/Jl79KvEIHQ+ARDxFD7KHY3nSokRi0=";
+
+  subPackages = [ "cmd/src" ];
+
+  CGO_ENABLED = 0;
```
Comment (IvarWithoutBones) [inline pkgs/development/tools/misc/src-cli/default.nix:21]: If i recall correctly this has the same result, but looks a bit nicer
```suggestion
  CGO_ENABLED = false;
```

File: pkgs/development/tools/misc/src-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "src-cli";
+  version = "4.0.0";
+
+  src = fetchFromGitHub {
+    owner = "sourcegraph";
+    repo = pname;
```
Comment (IvarWithoutBones) [inline pkgs/development/tools/misc/src-cli/default.nix:12]: For readabilities sake its better to spell this out IMO
```suggestion
    repo = "src-cli";
```

**Review by drupol — CHANGES_REQUESTED** [review]
What's the status of this PR?

The initial comments are not yet addressed. Do you plan to dedicate a little bit of your time to update it ?

Also, make sure to update to the latest version of `src-cli`.

If you don't plan to do it, please close the PR.

Thanks in advance.

**Comment by figsoda** [comment]
#239021

**Comment by drupol** [comment]
Closed in favor of https://github.com/NixOS/nixpkgs/pull/239021

**Pull request closed without merging**